staging/src/k8s.io/apiserver/pkg/endpoints/filters/authorization.go

```diff
@@ -47,14 +47,14 @@ func WithAuthorization(handler http.Handler, requestContextMapper request.Reques
 			return
 		}
 		authorized, reason, err := a.Authorize(attributes)
-		if authorized == authorizer.DecisionAllow {
```

Comment:
this might have been intentional to avoid leaking authorization errors to attackers. We better stay calm in those cases. /cc @liggitt

Reply:
allowed governs. an authorizer like RBAC could encounter evaluation errors and still allow the request (if some bindings didn't resolve, but other bindings did resolve and allowed the operation)

Reply:
@sttts @liggitt OK, should we add a comment here to give a hint?

Reply:
I agree this is a strange behaviour of a func with an error type. Something like `Authorize(a) (bool, string, Warnings, error)` with `type Warning []error` would be much more idiomatic.